runtime/src/pcr.rs

```diff
@@ -0,0 +1,124 @@
+// Licensed under the Apache-2.0 license
+
+use crate::Drivers;
+use caliptra_common::mailbox_api::{
+    ExtendPcrReq, IncrementPcrResetCounterReq, MailboxResp, MailboxRespHeader, QuotePcrsReq,
+    QuotePcrsResp,
+};
+use caliptra_drivers::{
+    cprintln, hand_off::DataStore, CaliptraError, CaliptraResult, Ecc384PrivKeyIn, KeyReadArgs,
+    PcrBank, PcrId,
+};
+use zerocopy::{transmute, FromBytes};
+
+pub struct PcrResetCounter {
+    counter: [u32; 32],
+}
+
+impl Default for PcrResetCounter {
+    fn default() -> Self {
+        PcrResetCounter::new()
+    }
+}
+
+impl PcrResetCounter {
+    fn new() -> PcrResetCounter {
+        PcrResetCounter { counter: [0; 32] }
+    }
+
+    pub fn get(&self, id: PcrId) -> u32 {
+        self.counter[usize::from(id)]
+    }
+
+    /// Increment the selected reset counter.
+    /// Returns `false` in case of a counter overflow
+    pub fn increment(&mut self, id: PcrId) -> bool {
+        let old_value = self.counter[usize::from(id)];
+
+        if let Some(new_value) = old_value.checked_add(1) {
+            self.counter[usize::from(id)] = new_value;
+
+            true
+        } else {
+            false
+        }
+    }
+}
+
+pub struct IncrementPcrResetCounter;
+impl IncrementPcrResetCounter {
+    pub(crate) fn execute(drivers: &mut Drivers, cmd_args: &[u8]) -> CaliptraResult<MailboxResp> {
+        let cmd = IncrementPcrResetCounterReq::read_from(cmd_args)
+            .ok_or(CaliptraError::RUNTIME_INSUFFICIENT_MEMORY)?;
+
+        let index =
+            u8::try_from(cmd.index).map_err(|_| CaliptraError::RUNTIME_MAILBOX_INVALID_PARAMS)?;
+
+        let pcr =
+            PcrId::try_from(index).map_err(|_| CaliptraError::RUNTIME_MAILBOX_INVALID_PARAMS)?;
+
+        if !drivers.pcr_reset.increment(pcr) {
+            return Err(CaliptraError::RUNTIME_INTERNAL);
```

Comment:
This PR should only implement the PCR_EXTEND feature.

But since this code relies on #1104, I rebased it on this branch because it has not been merged yet. I am not the author of the pcr reset code, so I think a review on this code fits better into into PR #1104